crates/bitwarden-crypto/src/enc_string/symmetric.rs

```diff
@@ -200,11 +232,45 @@ impl EncString {
         Ok(EncString::AesCbc256_HmacSha256_B64 { iv, mac, data })
     }
 
+    pub(crate) fn encrypt_xchacha20_poly1305(
+        data_dec: &[u8],
+        key: &XChaCha20Poly1305Key,
+    ) -> Result<EncString> {
+        let mut protected_header = coset::HeaderBuilder::new().build();
+        protected_header.alg = Some(coset::Algorithm::PrivateUse(cose::XCHACHA20_POLY1305));
```

Comment:
shouldn't this include a key id?

Also, area we limiting ourselves by defining `alg` === encryption type? I guess it just means we need to define a new private use alg if something changes... If this is true, we should define a constant map between `EncString::XChaCha20_Poly1305_Cose_B64` and `cose::XCHACHA20_POLY1305` so that matches can use a consistent state constant in their closures. This would probably be something like the `enc_type` function

Reply:
One key = one encryption algorithm. If anything changes in that regard, we use a new alg id, and a new symmetric key needs to be sampled.

However, I'm not sure whether we'd want a new encstring type, or whether we map all cose messages to type 7... Let's discuss this further synchronously?

Reply:
I don't think we ever came to a conclusion on this, all we agreed on is that changing an alg either can or cannot bump enc string types, it's our choice.

Reply:
Appending on the relevant info from https://github.com/bitwarden/sdk-internal/pull/181#discussion_r2031812104:

_If_ encryption type implies a specific alg rather than a an encoding, we need to enforce that equivalence on the decrypt side and fail should they not match up. Hypothetically that would mean `EncString::XChaCha20_Poly1305_Cose_B64` without `cose::XCHACHA20_POLY1305` alg would need to result in a decryption error